shared/fs/container.js

```diff
@@ -12,28 +12,55 @@ type OwnProps = {
 
 type StateProps = {
   path: Types.Path,
-  items: I.List<Types.Path>,
+  itemNames: I.List<string>,
   progress: Types.ProgressType,
+  sortSetting: Types.SortSetting,
+  username?: string,
+  pathItems: I.Map<Types.Path, Types.PathItem>,
 }
 
-type DispatchProps = {}
+type _sortSettingSetter = (setting: Types._SortSetting) => void
+type DispatchProps = {
+  _sortSettingChangeFactory: (path: Types.Path) => _sortSettingSetter,
+}
 
 const mapStateToProps = (state: TypedState, {routeProps}: OwnProps) => {
   const path = Types.stringToPath(routeProps.get('path', Constants.defaultPath))
+  const sortSetting = state.fs.pathUserSettings.get(path, Constants.makePathUserSetting()).get('sort')
   const itemDetail = state.fs.pathItems.get(path)
-  const items = itemDetail && itemDetail.type === 'folder' ? itemDetail.get('children', I.List()) : I.List()
+  const itemNames =
+    itemDetail && itemDetail.type === 'folder' ? itemDetail.get('children', I.List()) : I.List()
   const progress: Types.ProgressType = itemDetail ? itemDetail.progress : 'pending'
-  return {items, path, progress}
+  const username = state.config.username || undefined
+  const pathItems = state.fs.pathItems
+  return {itemNames, path, progress, sortSetting, username, pathItems}
 }
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
   _loadFolderList: (path: Types.Path) => dispatch(FsGen.createFolderListLoad({path})),
+  _sortSettingChangeFactory: (path: Types.Path) => (setting: Types.SortSetting) =>
+    dispatch(FsGen.createSortSetting({path: path, sortSetting: Constants.makeSortSetting(setting)})),
 })
 
-const mergeProps = ({path, items, progress}: StateProps, dispatchProps: DispatchProps, ownProps) => ({
-  items: items.map(name => Types.pathConcat(path, Types.pathToString(name))).toArray(),
+const mergeProps = (
+  {path, itemNames, progress, sortSetting, pathItems, username}: StateProps,
```

Comment:
lets not destructure these, just keep dispatchProps and stateProps

Reply:
@chrisnojima Got it. Although I'm replacing `_sortSettingChangeFactory` in `dispatchProps` with `setSortSetting`. Is it OK to leave the factory in there in final props? If not, what's the best way to do it here?

Reply:
only pass the props your component directly needs

Reply:
So instead of using the spread syntax on `dispatchProps` I should just explicitly list them out? Sounds good. (let me know if I got it wrong)

Reply:
sorry, i mean just use a regular variable
dispatchProps: DispatchProps in the mergeProps params

Reply:
Right; I mean below when constructing the props returned by `mergeProps`, we are currently using `...dispatchProps` to get everything from `dispatchProps`. I was asking if we should simply do `dispatchProps._loadFolderList` etc. there.

Reply:
ah, 
so i general i really dislike spreading any props in mergeProps. The problem with this approach is that you can get props to creep in and plumbed down that you wouldn't expect. This is especially true with ownProps (especially since the route tree will inject props into there). I think its better to be explicit and choose which props come down. (also keeps the _named things out). just makes things cleaner in my mind

Reply:
Got it; that's pretty clear. Thanks for explaining!